### PR TITLE
Process / return a new typed Lang class in i18n methods

### LINKED ISSUE #151: Define a type for language info
`i18n.get_language_details()` returns a `dict` which is not optimal in typed environments

### PR DESCRIPTION
**Pull request by benoit74**
Fix #151 

Changes:
- new `i18n.Lang` class to hold returned values
- `i18n.get_language_details()`, `i18n.get_iso_lang_data()`, `i18n.find_language_names()` and `i18n.update_with_macro` now process / return a new typed `Lang` class
- small fixes around renaming of `i18n.NotFound` to `i18n.NotFoundError` from previous PR (forgot changelog + few wrong docstrings)

### CODE REVIEW

**Review by rgaudin — CHANGES_REQUESTED** [review]
Welcomed change!

I see a missed opportunity for a proper type here that's not a wrapper around some dict.
If you're attached to that dict-ness, then subclass dict instead ; that'd be cleaner.

**Comment by benoit74** [comment]
The dict-ness is just the least effort path. Code is regularly updating / merging multiple properties at once, and it is a mess / very verbose to re-implement without the dict facilities (see e.g. `update_with_macro` with I don't know how to reimplement with a proper type). Since the main goal was to expose something more useful rather than a problem of maintainability, I considered that dict-ness was the appropriate solution.

That being that, I would also prefer to use a proper type if this does not makes the code too verbose. Shall we use pydantic BaseModel as base class? It would be a new dependency, it would help a bit, even if it would probably not solve all issues

**Comment by benoit74** [comment]
(and using a proper type was also my initial trial, but I quickly realized it was not that convenient to write)

**Comment by rgaudin** [comment]
Well I think you did not read my comment to the end and that none of your arguments make sense but given it was late when you replied, I think fresh eyes will have us agree that what I commited is simpler and has no downsides. Revert if that's not the case. I only tested via running the unit tests

**Comment by benoit74** [comment]
I did read it, just I'm not that attached to `dict`, and as said I would have preferred to use a proper type that does not subclass `dict`.

Let's go with `dict`, I will learn how to do it with a proper type next time (and maybe it is just not "wishable"), I like your simplification anyway.

I'll just replace `dict` by `Dict` to prepare for the future.

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - **BREAKING** `creator.Creator.add_metadata` and `creator.Creator.validate_metadata` now only accepts `bytes | str` as value (it must have been converted before call)
 - **BREAKING** second argument of `creator.Creator.add_metadata` has been renamed to `value` instead of `content` to align with other methods
 - When a type issue arises in metadata checks, wrong value type is displayed in exception
+- **BREAKING** `i18n.get_language_details()`, `i18n.get_iso_lang_data()`, `i18n.find_language_names()` and `i18n.update_with_macro` now process / return a new typed `Lang` class #151
+- **BREAKING** Rename `i18.NotFound` to `i18n.NotFoundError`
```
[the remark this answers is not in the document]
Comment (benoit74) [reply CHANGELOG.md:35]: Done

**Comment by benoit74** [comment]
Edit: ~~I'll just replace dict by Dict to prepare for the future~~ `dict` is already the future ^^

**Review by rgaudin — APPROVED** [review]

**Pull request merged**